### ISSUE TITLE
Database Error in model stg_exposure_relationships

### ISSUE THREAD

**Issue by Lorenitz** [dbt-labs/dbt-project-evaluator#145]
### Describe the bug
Error message when executing the command:
`dbt build --select package:dbt_project_evaluator`

### Steps to reproduce
`dbt build --select package:dbt_project_evaluator`

### Actual results

Completed Succesfully

### Screenshots and log output
![Screenshot 2022-06-19 at 21 45 46](https://user-images.githubusercontent.com/31211443/174497920-3539157e-4ea7-4496-8027-c40f7ded5eed.png)

### System information
```
packages:
  - package: dbt-labs/dbt_utils
    version: 0.8.5
  - package: tnightengale/dbt_meta_testing
    version: 0.3.5
  - package: Snowflake-Labs/dbt_constraints
    version: [">=0.3.0", "<0.4.0"]
  - git: "https://github.com/dbt-labs/dbt-project-evaluator.git"
    revision: v0.1.0
```

**Which database are you using dbt with?**
- [x] postgres
- [x] redshift
- [ ] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
Core:
  - installed: 1.1.0
  - latest:    1.1.1 - Update available!

  Your version of dbt-core is out of date!
  You can find instructions for upgrading here:
  https://docs.getdbt.com/docs/installation

Plugins:
  - redshift: 1.1.0 - Up to date!
  - postgres: 1.1.0 - Update available!
```

**Comment by b-per** [comment]
Thanks for the report @Lorenitz 

Would you be able to share the SQL that was run for the 3 queries that failed? You should be able to find it in the logs or under `target/run/`

**Comment by Lorenitz** [comment]
Hello, Thank you for replying my message :)
![Screenshot 2022-06-21 at 13 43 44](https://user-images.githubusercontent.com/31211443/174791391-796a6c38-7a18-42d2-942e-13af492fa4a5.png)

Should I check the file int_all_graph_resources.sql ?

**Comment by Lorenitz** [comment]
If yes, that is the sql for this file:


 ```
 create  table
    "data"."z_odetriche_test"."int_all_graph_resources__dbt_tmp"
    
    
    
  as (
    -- one row for each resource in the graph
with unioned as (

    

        (
            select

                cast('"data"."z_odetriche_test"."stg_nodes"' as varchar) as _dbt_source_relation,
                
                    cast("unique_id" as character varying(256)) as "unique_id" ,
                    cast("name" as character varying(256)) as "name" ,
                    cast("resource_type" as character varying(256)) as "resource_type" ,
                    cast("file_path" as character varying(256)) as "file_path" ,
                    cast("is_enabled" as boolean) as "is_enabled" ,
                    cast("materialized" as character varying(11)) as "materialized" ,
                    cast("on_schema_change" as character varying(6)) as "on_schema_change" ,
                    cast("database" as character varying(4)) as "database" ,
                    cast("schema" as character varying(32)) as "schema" ,
                    cast("package_name" as character varying(256)) as "package_name" ,
                    cast("alias" as character varying(256)) as "alias" ,
                    cast("is_described" as boolean) as "is_described" ,
                    cast("column_name" as character varying(44)) as "column_name" ,
                    cast(null as character varying(256)) as "exposure_type" ,
                    cast(null as character varying(256)) as "maturity" ,
                    cast(null as character varying(256)) as "url" ,
                    cast(null as character varying(256)) as "metric_type" ,
                    cast(null as character varying(256)) as "model" ,
                    cast(null as character varying(256)) as "label" ,
                    cast(null as character varying(256)) as "sql" ,
                    cast(null as character varying(256)) as "timestamp" ,
                    cast(null as character varying(32)) as "source_name" ,
                    cast(null as boolean) as "is_source_described" ,
                    cast(null as character varying(17)) as "loaded_at_field" ,
                    cast(null as character varying(8)) as "loader" ,
                    cast(null as character varying(47)) as "identifier" 

            from "data"."z_odetriche_test"."stg_nodes"

            
        )

        union all
        

        (
            select

                cast('"data"."z_odetriche_test"."stg_exposures"' as varchar) as _dbt_source_relation,
                
                    cast("unique_id" as character varying(256)) as "unique_id" ,
                    cast("name" as character varying(256)) as "name" ,
                    cast("resource_type" as character varying(256)) as "resource_type" ,
                    cast("file_path" as character varying(256)) as "file_path" ,
                    cast(null as boolean) as "is_enabled" ,
                    cast(null as character varying(11)) as "materialized" ,
                    cast(null as character varying(6)) as "on_schema_change" ,
                    cast(null as character varying(4)) as "database" ,
                    cast(null as character varying(32)) as "schema" ,
                    cast("package_name" as character varying(256)) as "package_name" ,
                    cast(null as character varying(256)) as "alias" ,
                    cast("is_described" as boolean) as "is_described" ,
                    cast(null as character varying(44)) as "column_name" ,
                    cast("exposure_type" as character varying(256)) as "exposure_type" ,
                    cast("maturity" as character varying(256)) as "maturity" ,
                    cast("url" as character varying(256)) as "url" ,
                    cast(null as character varying(256)) as "metric_type" ,
                    cast(null as character varying(256)) as "model" ,
                    cast(null as character varying(256)) as "label" ,
                    cast(null as character varying(256)) as "sql" ,
                    cast(null as character varying(256)) as "timestamp" ,
                    cast(null as character varying(32)) as "source_name" ,
                    cast(null as boolean) as "is_source_described" ,
                    cast(null as character varying(17)) as "loaded_at_field" ,
                    cast(null as character varying(8)) as "loader" ,
                    cast(null as character varying(47)) as "identifier" 

            from "data"."z_odetriche_test"."stg_exposures"

            
        )

        union all
        

        (
            select

                cast('"data"."z_odetriche_test"."stg_metrics"' as varchar) as _dbt_source_relation,
                
                    cast("unique_id" as character varying(256)) as "unique_id" ,
                    cast("name" as character varying(256)) as "name" ,
                    cast("resource_type" as character varying(256)) as "resource_type" ,
                    cast("file_path" as character varying(256)) as "file_path" ,
                    cast(null as boolean) as "is_enabled" ,
                    cast(null as character varying(11)) as "materialized" ,
                    cast(null as character varying(6)) as "on_schema_change" ,
                    cast(null as character varying(4)) as "database" ,
                    cast(null as character varying(32)) as "schema" ,
                    cast("package_name" as character varying(256)) as "package_name" ,
                    cast(null as character varying(256)) as "alias" ,
                    cast("is_described" as boolean) as "is_described" ,
                    cast(null as character varying(44)) as "column_name" ,
                    cast(null as character varying(256)) as "exposure_type" ,
                    cast(null as character varying(256)) as "maturity" ,
                    cast(null as character varying(256)) as "url" ,
                    cast("metric_type" as character varying(256)) as "metric_type" ,
                    cast("model" as character varying(256)) as "model" ,
                    cast("label" as character varying(256)) as "label" ,
                    cast("sql" as character varying(256)) as "sql" ,
                    cast("timestamp" as character varying(256)) as "timestamp" ,
                    cast(null as character varying(32)) as "source_name" ,
                    cast(null as boolean) as "is_source_described" ,
                    cast(null as character varying(17)) as "loaded_at_field" ,
                    cast(null as character varying(8)) as "loader" ,
                    cast(null as character varying(47)) as "identifier" 

            from "data"."z_odetriche_test"."stg_metrics"

            
        )

        union all
        

        (
            select

                cast('"data"."z_odetriche_test"."stg_sources"' as varchar) as _dbt_source_relation,
                
                    cast("unique_id" as character varying(256)) as "unique_id" ,
                    cast("name" as character varying(256)) as "name" ,
                    cast("resource_type" as character varying(256)) as "resource_type" ,
                    cast("file_path" as character varying(256)) as "file_path" ,
                    cast("is_enabled" as boolean) as "is_enabled" ,
                    cast(null as character varying(11)) as "materialized" ,
                    cast(null as character varying(6)) as "on_schema_change" ,
                    cast("database" as character varying(4)) as "database" ,
                    cast("schema" as character varying(32)) as "schema" ,
                    cast("package_name" as character varying(256)) as "package_name" ,
                    cast("alias" as character varying(256)) as "alias" ,
                    cast("is_described" as boolean) as "is_described" ,
                    cast(null as character varying(44)) as "column_name" ,
                    cast(null as character varying(256)) as "exposure_type" ,
                    cast(null as character varying(256)) as "maturity" ,
                    cast(null as character varying(256)) as "url" ,
                    cast(null as character varying(256)) as "metric_type" ,
                    cast(null as character varying(256)) as "model" ,
                    cast(null as character varying(256)) as "label" ,
                    cast(null as character varying(256)) as "sql" ,
                    cast(null as character varying(256)) as "timestamp" ,
                    cast("source_name" as character varying(32)) as "source_name" ,
                    cast("is_source_described" as boolean) as "is_source_described" ,
                    cast("loaded_at_field" as character varying(17)) as "loaded_at_field" ,
                    cast("loader" as character varying(8)) as "loader" ,
                    cast("identifier" as character varying(47)) as "identifier" 

            from "data"."z_odetriche_test"."stg_sources"

            
        )

        

),

naming_convention_prefixes as (
    select * from "data"."z_odetriche_test"."stg_naming_convention_prefixes"
), 

naming_convention_folders as (
    select * from "data"."z_odetriche_test"."stg_naming_convention_folders"
), 

unioned_with_calc as (
    select 
        *,
        case 
            when resource_type = 'source' then  source_name || '.' || name
            else name 
        end as resource_name,
        case
            when resource_type = 'source' then null
            else 

  
    

    split_part(
        name,
        '_',
        1
        )


  

||'_' 
        end as prefix,
        

    replace(
        file_path,
        regexp_replace(file_path,'.*/',''),
        ''
    )
    

 as directory_path,
        regexp_replace(file_path,'.*/','') as file_name 
    from unioned
    where coalesce(is_enabled, True) = True and package_name != 'dbt_project_evaluator'
), 

joined as (

    select
        unioned_with_calc.unique_id as resource_id, 
        unioned_with_calc.resource_name, 
        unioned_with_calc.prefix, 
        unioned_with_calc.resource_type, 
        unioned_with_calc.file_path, 
        unioned_with_calc.directory_path,
        unioned_with_calc.file_name,
        case 
            when unioned_with_calc.resource_type in ('test', 'source', 'metric', 'exposure', 'seed') then null
            else naming_convention_prefixes.model_type 
        end as model_type_prefix,
        case 
            when unioned_with_calc.resource_type in ('test', 'source', 'metric', 'exposure', 'seed') then null
            when 

    position(
        naming_convention_folders.folder_name_value in unioned_with_calc.directory_path
    ) = 0 then null
            else naming_convention_folders.model_type 
        end as model_type_folder,
        

    position(
        naming_convention_folders.folder_name_value in unioned_with_calc.directory_path
    ) as position_folder,  
        nullif(unioned_with_calc.column_name, '') as column_name,
        unioned_with_calc.resource_name like 'unique%' and unioned_with_calc.resource_type = 'test' as is_not_null_test,
        unioned_with_calc.resource_name like 'not_null%' and unioned_with_calc.resource_type = 'test' as is_unique_test,
        unioned_with_calc.is_enabled, 
        unioned_with_calc.materialized, 
        unioned_with_calc.on_schema_change, 
        unioned_with_calc.database, 
        unioned_with_calc.schema, 
        unioned_with_calc.package_name, 
        unioned_with_calc.alias, 
        unioned_with_calc.is_described, 
        unioned_with_calc.exposure_type, 
        unioned_with_calc.maturity, 
        unioned_with_calc.url, 
        unioned_with_calc.metric_type, 
        unioned_with_calc.model, 
        unioned_with_calc.label, 
        unioned_with_calc.sql, 
        unioned_with_calc.timestamp as timestamp,  
        unioned_with_calc.source_name, -- NULL for non-source resources
        unioned_with_calc.is_source_described, 
        unioned_with_calc.loaded_at_field, 
        unioned_with_calc.loader, 
        unioned_with_calc.identifier

    from unioned_with_calc
    left join naming_convention_prefixes
        on unioned_with_calc.prefix = naming_convention_prefixes.prefix_value

    cross join naming_convention_folders   

), 

calculate_model_type as (
    select 
        *, 
        case 
            when resource_type in ('test', 'source', 'metric', 'exposure', 'seed') then null
            -- by default we will define the model type based on its prefix in the case prefix and folder types are different
            else coalesce(model_type_prefix, model_type_folder, 'other') 
        end as model_type,
        row_number() over (partition by resource_id order by position_folder desc) as folder_name_rank
    from joined
),

final as (
    select
        *
    from calculate_model_type
    where folder_name_rank = 1
)

select 
    *
from final
  );
```

**Comment by b-per** [comment]
Thanks! I have checked this code and it worked on Redshift my side, which means that the issue is with the upstream `stg_` models that are materialized as views.

For now, could you provide the same extract from run but for the model `stg_exposure_relationships` or `stg_metric_relationships`?

**Comment by Lorenitz** [comment]
Thank you very much! I am going to try :)

**Comment by b-per** [comment]
This is now closed in `main` with https://github.com/dbt-labs/dbt-project-evaluator/issues/137 being merged.
We will include the fix in the release `0.1.1`
